app/controllers/api/custom_buttons_controller.rb

```diff
@@ -4,20 +4,24 @@ def create_resource(_type, _id, data)
       custom_button = CustomButton.new(data.except("resource_action", "options"))
       custom_button.userid = User.current_user.userid
       custom_button.options = data["options"].deep_symbolize_keys if data["options"]
-      custom_button.visibility = data["visibility"].deep_symbolize_keys if data["visibility"]
-      custom_button.resource_action = find_or_create_resource_action(data["resource_action"]) if data.key?("resource_action")
-      if custom_button.save
-        custom_button
-      else
-        raise BadRequestError, "Failed to create new custom button - #{custom_button.errors.full_messages.join(", ")}"
-      end
+      custom_button.save!
+      custom_button.resource_action.update!(data["resource_action"].deep_symbolize_keys) if data.key?("resource_action")
+      custom_button.save!
```

Comment:
Maybe these should be wrapped in a transaction, so that we can rollback on exception?

Reply:
Fixed :)